Hestia/HestiaExtra.lua

```diff
@@ -3504,6 +3524,29 @@ if ModUtil ~= nil then
 		CreateAnimationsBetween({ Animation = "FistVacuumFx", DestinationId = victimId, Id = CurrentRun.Hero.ObjectId, Length = args.distanceBuffer, Stretch = true, UseZLocation = false, Group = "FX_Standing_Add" })
 		PlaySound({ Name = "/SFX/Player Sounds/ZagreusFistMagnetismVacuumActivate", Id = victimId })
 	end
+    
+
+	-- Hestia FreeHealthTrait
+	ModUtil.Path.Wrap( "SpawnRoomReward", 
+		function(baseFunc, eventSource, args )
+			--ModUtil.Hades.PrintStackChunks(ModUtil.ToString(CurrentRun.CurrentRoom.Encounter.EncounterType)) 
+			if (HeroHasTrait("FreeHealthTrait") and CurrentRun.CurrentRoom.Encounter ~= nil and CurrentRun.CurrentRoom.Encounter.EncounterType ~= nil and 
+				(CurrentRun.CurrentRoom.Encounter.EncounterType == "Boss" or CurrentRun.CurrentRoom.Encounter.EncounterType == "Miniboss" or CurrentRun.CurrentRoom.Encounter.EncounterType == "OptionalBoss")) then
```

Comment:
I can accept PR, but you'll need to create a new RoomRewardMaxHealthDrop like RoomRewardLessHealthDrop. that gives 15 hp instead of 25 hp for miniboss and optionalboss. Also, for the miniboss list we talked about, we want to tag those encounters and give them "CanGiveExtraReward = true" a custom variable to know when to give the extra reward.